app/models/piece.rb

```diff
@@ -2,6 +2,36 @@ class Piece < ActiveRecord::Base
   belongs_to :game
   belongs_to :player
 
+  def valid_move?(destination_x, destination_y)
+    game = self.get_game
+    # Check if piece is obstructed
+    if self.is_obstructed?(destination_x, destination_y) == true
+      return false
+    # Check if destination is occupied by a piece of the same color
+    elsif game.piece_present?(destination_x, destination_y) == true
```

Comment:
Can be lazy and remove the `== true`. `if bool` means if it's true, `if !bool` is false.